xsuite/src/cli/buildReproducibleAction.ts

```diff
@@ -0,0 +1,205 @@
+import { PathLike, existsSync, mkdirSync, readdirSync } from "fs";
+import path from "path";
+import { Command } from "commander";
+import { log } from "../_stdio";
+import {
+  defaultIgnore,
+  findBuildableContractDirs,
+  getGid,
+  getUid,
+} from "./buildUtils";
+import {
+  defaultReproducibleDockerImage,
+  logAndRunCommand,
+  logError,
+  promptUserWithRetry,
+} from "./helpers";
+
+export const registerBuildReproducibleCmd = (cmd: Command) => {
+  cmd
+    .command("build-reproducible")
+    .description("Build contract.")
+    .option(
+      "--image <IMAGE_TAG>",
+      "Specify the tag of the docker image, that is used to build the contract (e. g. multiversx/sdk-rust-contract-builder:v6.1.0)",
+    )
+    .option(
+      "--dir <DIR>",
+      "Directory in which the command is executed (default: $(PWD))",
+    )
+    .option(
+      "--target-dir <TARGET_DIR>",
+      "Target directory used by Rust compiler (default: $(PWD)/target)",
```

Comment:
`target` is the name of the Rust target folder, and so `target` for the output would most likely conflict with Rust regular `target` folder. `output-reproducible` would be a more telling default name.